fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorEvaluator.java

```diff
@@ -111,13 +112,17 @@ public Function getMetaFunction(FunctionName name, Type[] args) {
         return new Function(name, Lists.newArrayList(args), Type.VARCHAR, false);
     }
 
+    public ScalarOperator evaluation(CallOperator root) {
+        return evaluation(root, false);
+    }
+
     /**
      * evaluation a fe built-in function
      *
      * @param root CallOperator root
      * @return ConstantOperator if the CallOperator is effect (All child constant/FE builtin function support/....)
      */
-    public ScalarOperator evaluation(CallOperator root) {
+    public ScalarOperator evaluation(CallOperator root, boolean needMonotonic) {
```

Comment:
Which function needs `needMonotonic `? add some docs?

Reply:
When a function is a monotonic fucntion, which means for any value within a range, the first and last endpoints are the new extreme points after the function mapping. Then we can remapping the partition ranges using constant folding rule to obatin a new range to helps us prune paritions. I will add some comments